src/Compilers/CSharp/Test/Semantic/Semantics/StructConstructorTests.cs

```diff
@@ -0,0 +1,1329 @@
+            comp.VerifyDiagnostics(
+                // (3,22): error CS8918: The parameterless struct constructor must be 'public'.
+                // struct S3 { internal S3() { } }
+                Diagnostic(ErrorCode.ERR_NonPublicParameterlessStructConstructor, "S3").WithLocation(3, 22),
+                // (4,21): error CS8918: The parameterless struct constructor must be 'public'.
+                // struct S4 { private S4() { } }
+                Diagnostic(ErrorCode.ERR_NonPublicParameterlessStructConstructor, "S4").WithLocation(4, 21),
+                // (12,27): error CS1736: Default parameter value for 's' must be a compile-time constant
+                //     static void G2(S2 s = new()) { }
+                Diagnostic(ErrorCode.ERR_DefaultValueMustBeConstant, "new()").WithArguments("s").WithLocation(12, 27),
+                // (13,27): error CS1736: Default parameter value for 's' must be a compile-time constant
+                //     static void G3(S3 s = new()) { }
+                Diagnostic(ErrorCode.ERR_DefaultValueMustBeConstant, "new()").WithArguments("s").WithLocation(13, 27),
+                // (14,27): error CS0122: 'S4.S4()' is inaccessible due to its protection level
+                //     static void G4(S4 s = new()) { }
+                Diagnostic(ErrorCode.ERR_BadAccess, "new()").WithArguments("S4.S4()").WithLocation(14, 27),
+                // (14,27): error CS1736: Default parameter value for 's' must be a compile-time constant
+                //     static void G4(S4 s = new()) { }
+                Diagnostic(ErrorCode.ERR_DefaultValueMustBeConstant, "new()").WithArguments("s").WithLocation(14, 27));
+        }
+
+        [Fact]
+        public void ParameterDefaultValues_02()
+        {
+            var source =
+@"struct S1
+{
+    object X = 1;
+}
+struct S2
+{
+    object X = 2;
+    public S2() { }
+}
+struct S3
+{
+    object X = 3;
+    public S3(object x) { X = x; }
+}
+class Program
+{
+    static void F1(S1 s = default) { }
+    static void F2(S2 s = default) { }
+    static void F3(S3 s = default) { }
+    static void G1(S1 s = new()) { }
+    static void G2(S2 s = new()) { }
+    static void G3(S3 s = new()) { }
+}";
+            var comp = CreateCompilation(source, parseOptions: TestOptions.RegularPreview);
+            comp.VerifyDiagnostics(
+                // (20,27): error CS1736: Default parameter value for 's' must be a compile-time constant
+                //     static void G1(S1 s = new()) { }
+                Diagnostic(ErrorCode.ERR_DefaultValueMustBeConstant, "new()").WithArguments("s").WithLocation(20, 27),
+                // (21,27): error CS1736: Default parameter value for 's' must be a compile-time constant
+                //     static void G2(S2 s = new()) { }
+                Diagnostic(ErrorCode.ERR_DefaultValueMustBeConstant, "new()").WithArguments("s").WithLocation(21, 27));
+        }
+    }
+}
```

Comment:
Consider adding some tests with initializer values on properties, including getter-only properties (ctor should write directly to backing field).